src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/TenantApiController.cs

```diff
@@ -295,7 +295,7 @@ public async Task<IActionResult> Remove(string tenantName)
 
         [HttpPost]
         [Route("setup")]
-        public async Task<ActionResult> Setup(SetupApiViewModel model, [FromForm] IFormFile recipe = null)
```

Comment:
How this backward compatible while the old one accepts a file not a JSON string?

Reply:
The old one doesn't expect a file either. For that, as well as for the new one, the HTTP request is something like this, with the recipe passed in being part of the JSON body:

```json
{"name":"ApiClientTenant638565081195196341","siteName":"Api Client Tenant Site","databaseProvider":"Sqlite","connectionString":"","tablePrefix":"apiclienttenant638565081195196341","userName":"admin","email":"admin@example.com","password":"Password1!","recipeName":null,"Recipe":"{\r\n  \u0022name\u0022: \u0022Agency\u0022,}\r\n","siteTimeZone":"Europe/Budapest"}
```

Reply:
I'm curious why the `IFormFile` was there :)

Reply:
I won't be able to answer that.

Reply:
This parameter was added in [this PR](https://github.com/OrchardCMS/OrchardCore/pull/14058/files#diff-1f259b7c710f0b611bbf75b51687ee076368945796958a85ec12be8255f83462R298), but before that it was a property in `SetupApiViewModel` [ever since the tenant API was introduced](https://github.com/OrchardCMS/OrchardCore/commit/384e0a242278d4f5eff1b3d37a9a0402698aa176#diff-8aa46e781b521a5f305d6c0679550e305f369b53839ad38fe48ee22424102c70R32). So I guess moving it into a standalone parameter was just for the sake of adding `[FromForm]`. I don't see any related changes to moving from `SetupApiViewModel.Recipe` to `recipe` in #14058, so it's a safe bet that this optional pathway was dead code.